Adds errorCode attribute to API error response

OluwaninsolaAO opened the pull request.

OluwaninsolaAO committed:
Adds errorCode attribute to API error response

OluwaninsolaAO committed:
Adds ommited errorCode for Internal Server Error

OluwaninsolaAO committed:
Setups API throttler with a more enhanced API errorCode handling

OluwaninsolaAO committed:
Bugfixes: improves errorCode cross reference in exc filter

7H3XXX commented on packages/types/src/error.codes.ts:
```diff
@@ -0,0 +1,101 @@
+  ACCOUNT_VERIFICATION_REQUIRED: 'ACCOUNT_VERIFICATION_REQUIRED',
+
+  // --- Resource & Data Errors ---
+  RESOURCE_NOT_FOUND: 'RESOURCE_NOT_FOUND',
```
I approve the changes and like how structured your code is. For the error codes I only have one problem. Saying resource and data errors is a bit misleading cause when translating we won't be able to know which type of resources the error originated from. So I think it's better if you can name each one precisely or find a better alternative to make it easy to set keys for translation.

OluwaninsolaAO replied:
There for the review. ✨✨

Okay, so are you suggesting that there should be a unique error code for each resource/entities from the backend?

```
USER_NOT_FOUND
COMPANY_NOT_FOUND
JOB_LISTING_NOT_FOUND
...
```

7H3XXX replied:
Yes or a better and scalable alternative to make translation on the frontend simpler. Some browsing or LLM can help pick the right option.

OluwaninsolaAO replied:
Okay, do let me know if you have any references, but imma going to go with creating different error codes for each resource/entities.

OluwaninsolaAO committed:
Creates additional API resources ---

[x] My Organisations API resrouces for employers
[x] Public Organisations API for looking up organisation details
[x] A piece of Manage Organisation API resource for app admin areas
[x] A couple of changes to User Service and
[x] Updates Minio Controller with a better filename handling

Maurrreenn reviewed (approved):
Seen, although I haven't looked much at it, to be discussed later then??
